src/blockchain/esplora/mod.rs

```diff
@@ -0,0 +1,42 @@
+//! Esplora
+//!
+//! This module defines a [`EsploraBlockchain`] struct that can query an Esplora
+//! backend populate the wallet's [database](crate::database::Database) by:
+//!
+//! ## Example
+//!
+//! ```no_run
+//! # use bdk::blockchain::esplora::EsploraBlockchain;
+//! let blockchain = EsploraBlockchain::new("https://blockstream.info/testnet/api");
+//! # Ok::<(), bdk::Error>(())
+//! ```
+//!
+//! Esplora blockchain can use either `ureq` or `reqwest` for the HTTP client
+//! depending on your needs (blocking or async respectively).
+//!
+//! Please note, to configure the Esplora HTTP client correctly use one of:
+//! Blocking:  --features esplora,ureq
+//! Async:     --features async-features
```

Comment:
This seems to imply that `async-features` is enough to enable the esplora async module, but I think you actually need the explicit `esplora` feature as well.

Reply:
Well two things. Most importantly this comment uses the wrong feature flag, its `async-interface` not `async-features`, so thanks for flagging this line. With `--features async-interface`, if I understand what I've done correctly, there is no need for explicit `esplora`. This was intentional, since there is only one async backend supported I figured we could enable it with
```
async-interface = ["async-trait", "esplora", "reqwest"]
```
I've pushed another commit attempting to clarify this further in `Cargo.toml`.

Are you happy with this design?

Reply:
'Comfortable' is probably a better word that 'happy' - it still sucks that we cannot use `--all-features` :)

Reply:
I'm not so sure about this, even though I don't particularly like the async stuff today I'm not completely excluding that we can implement more async `Blockchain` in the future.

I'd rather have two separate features, one to enable the async interface (which means that the trait will have async methods) and a different one to enable esplora.

This would also allow external users to implement their own custom async `Blockchain` without having to pull in reqwest and all the esplora code.

Reply:
Oh, good point. I did not think about users extending `bdk` - interesting things to think about when hacking on this repo that I'm not used to thinking about. Will see what I can come up with, thanks.

Reply:
Made `esplora`/`reqwest` independent of `async-interface`. Update docs in manifest to further assist in understanding whats required when using these features.